### ISSUE TITLE
Ant-style path pattern support broken when `ktlint` is invoked from the command line

### ISSUE THREAD

**Issue by 0x6675636b796f75676974687562** [pinterest/ktlint#1601]
## Expected Behavior

Judging from its documentation, `ktlint` should support _Ant_-style path patterns, as described in

 * https://ant.apache.org/manual/dirtasks.html#patterns
 * https://www.jetbrains.com/help/teamcity/wildcards.html#Antlike+Wildcards
 * https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/util/AntPathMatcher.html

## Observed Behavior

Consider I have one or more `*.kt` files in the `src/` subdirectory of the current directory.

 * The invocation below works, the regular shell wildcard gets expanded by any POSIX-compliant shell:
    ```bash
    ktlint src/*.kt
    ```
 * Any of these invocations work in `zsh` (the more complex _Ant_-style patterns get expanded by the shell itself):
   ```zsh
   ktlint **/*.kt
   ktlint ./**/*.kt
   ktlint src/**/*.kt
   ktlint ./src/**/*.kt
   ktlint **/src/*.kt
   ktlint ./**/src/*.kt
   ktlint **/src/**/*.kt
   ktlint ./**/src/**/*.kt
   ```

Now, let's enclose the last argument in double quotes so that it gets expanded (as we hope for) by `ktlint` rather than the underlying shell:

 * `ktlint "src/*.kt"` &mdash; doesn't work (`No files matched [src/*.kt]`).
 * `ktlint "**/*.kt"` &mdash; works.
 * `ktlint "./**/*.kt"` &mdash; doesn't work.
 * `ktlint "src/**/*.kt"` &mdash; doesn't work (this is actually the example from the output of  `ktlint --help`).
 * `ktlint "./src/**/*.kt"` &mdash; doesn't work.
 * `ktlint "**/src/*.kt"` &mdash; doesn't work.
 * `ktlint "./**/src/*.kt"` &mdash; doesn't work.
 * `ktlint "**/src/**/*.kt"` &mdash; doesn't work.
 * `ktlint "./**/src/**/*.kt"` &mdash; doesn't work.

## Steps to Reproduce

 * Download the fat JAR from https://github.com/pinterest/ktlint/releases/tag/0.47.0.
 * Run the far JAR against any non-empty codebase, using either a POSIX shell or `cmd.exe`:
    ```bash
    java -Xmx512m --add-opens java.base/java.lang=ALL-UNNAMED -jar ktlint <path pattern(s)>
    ```

## Your Environment

* Version of ktlint used: **0.47**.
* Relevant parts of the `.editorconfig` settings: none
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): **CLI**
* Version of Gradle used (if applicable): N/A
* Operating System and version:
  * `MINGW64_NT-10.0-19043 unit-725 3.3.4-341.x86_64 2022-05-09 11:56 UTC x86_64 Msys`
  * `Linux unit-725 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64 GNU/Linux`

**Comment by 0x6675636b796f75676974687562** [comment]
An additional problem is that path patterns which ktlint produces automatically, when launched w/o arguments, are **platform-specific**, i. e. they use backslashes on Windows: #1611.

This is a clear violation of Ant pattern syntax (see the linked pages).

**Comment by paul-dingemans** [comment]
Yes, apparently I have just assumed that the path patterns followed the default directory separator of the file system.

```
internal val defaultPatterns = setOf(
    "**$globSeparator*.kt",
    "**$globSeparator*.kts"
)

...

private val globSeparator: String get() =
    when {
        os.startsWith("windows", ignoreCase = true) -> "\\\\"
        else -> "/"
    }
```

**Comment by paul-dingemans** [comment]
Hi @0x6675636b796f75676974687562 ,

As far as I can see, I have fixed the problem with path traversing (#1600) and the ant-path parsing. I would be grateful if you could test latest snapshot (see https://pinterest.github.io/ktlint/install/snapshot-build/) to see whether the problems are resolved on Windows.

**Comment by 0x6675636b796f75676974687562** [comment]
@paul-dingemans, I've tested the latest snapshot on Windows, and all the aforementioned _Ant_ patterns work now.

Thank you!
